src/test/java/core/basesyntax/services/FruitMapperTest.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class FruitMapperTest {
+    private static FruitMapper fruitMapper;
+    private static List<String> emptyList;
+    private static Map<List<FruitTransaction>, List<String>> correctValues;
+    private static List<String> wrongLength;
+    private static List<String> wrongOperatorFruitOperation;
+
+    @BeforeAll
+    static void beforeAll() {
+        fruitMapper = new FruitMapperImpl();
+        emptyList = new ArrayList<>();
+        correctValues = Map.of(
+                List.of(new FruitTransaction(Operation.BALANCE, "banana", 20)),
+                List.of("b,banana,20"),
+                List.of(new FruitTransaction(Operation.PURCHASE, "apple", 20)),
+                List.of("p,apple,20"),
+                List.of(new FruitTransaction(Operation.SUPPLY, "banana", 100)),
+                List.of("s,banana,100"),
+                List.of(new FruitTransaction(Operation.RETURN, "apple", 10)),
+                List.of("r,apple,10"));
+
+        wrongOperatorFruitOperation = List.of("a,kiwi,20");
+        wrongLength = List.of("a, apple, 100, 100");
+    }
+
+    @Test
+    void mapDate_correct_OK() {
+        for (Map.Entry<List<FruitTransaction>, List<String>> map : correctValues.entrySet()) {
+            List<FruitTransaction> excepted = map.getKey();
+            List<String> value = map.getValue();
+            List<FruitTransaction> actual = fruitMapper.mapData(value);
+            assertEquals(excepted, actual);
+        }
+    }

```

Comment:
```suggestion
 @ParameterizedTest
    @MethodSource("getCorrectValues")
    void mapData_correct_OK(List<FruitTransaction> expected, List<String> value) {
        List<FruitTransaction> actual = fruitMapper.mapData(value);
        assertEquals(expected, actual);
    }

    private static Stream<Arguments> getCorrectValues() {
        return correctValues.entrySet().stream()
            .map(entry -> Arguments.of(entry.getKey(), entry.getValue()));
    }
```

Reply:
For tests with different parameters we should use [parameterized tests](https://medium.com/@reetesh043/junit-5-parameterized-tests-7e2af541fa2) (non cycles).